api/app.go

```diff
@@ -97,15 +97,26 @@ func (a *App) getRouter() *mux.Router {
 		&VersionMiddleware{},
 	)).Methods("GET").Name("offer_templates")
 
-	r.Handle("/templates", Chain(
+	r.Handle("/templates/{id}/insert", Chain(
 		&OfferTemplateHandler{App: a, Method: "insert"},
 		&NewRelicMiddleware{App: a},
 		&AuthMiddleware{App: a},
 		&LoggingMiddleware{App: a},
 		&VersionMiddleware{},
+		NewParamKeyMiddleware(a, govalidator.IsUUIDv4),
 		NewValidationMiddleware(func() interface{} { return &models.OfferTemplate{} }),
 	)).Methods("POST").Name("offer_templates")
 
+	r.HandleFunc("/templates/{id}/update", Chain(
```

Comment:
Here we're creating a new version for the template with a given key, correct?

I'd use PUT /templates/{key} or something like this